### PR TITLE
tests: Enable progress notify in linearizability tests

### PR DESCRIPTION
**Pull request by serathius**
cc @ahrtr @ptabor

### CODE REVIEW

File: tests/linearizability/linearizability_test.go [hunk 1]
```diff
@@ -177,7 +179,7 @@ func TestLinearizability(t *testing.T) {
 				waitBetweenTriggers: waitBetweenFailpointTriggers,
 			}, *scenario.traffic)
 			forcestopCluster(clus)
-			validateWatchResponses(t, watchResponses)
+			validateWatchResponses(t, watchResponses, clus.Cfg.WatchProcessNotifyInterval != 0)
```
Comment (fuweid) [inline tests/linearizability/linearizability_test.go:182]: Not sure I understand it correctly.
.
By default, the interval is 10 minutes. Currently, the test case won't hit the limit.
So `x != 0` can be used as disabled progress-notify? is that correct?
If so, is it possible to add some comment for this?

Comment (serathius) [reply tests/linearizability/linearizability_test.go:182]: > So x != 0 can be used as disabled progress-notify?

No, in e2e cluster config 0 is empty value which means, don't set. In that case etcd will use the default. As you mentioned the default interval is 10 minutes which is too much for test takes around 5 seconds.

Comment (fuweid) [reply tests/linearizability/linearizability_test.go:182]: Thanks for the comment!

**Review by fuweid — APPROVED** [review]
LGTM

**Comment by serathius** [comment]
As per @logicalhan suggestion I have added validation that there at least one progress notify that returns same revision as previous watch event.

File: tests/linearizability/linearizability_test.go [hunk 2]
```diff
@@ -177,7 +179,8 @@ func TestLinearizability(t *testing.T) {
 				waitBetweenTriggers: waitBetweenFailpointTriggers,
 			}, *scenario.traffic)
 			forcestopCluster(clus)
-			validateWatchResponses(t, watchResponses)
+			watchProgressNotifyEnabled := clus.Cfg.WatchProcessNotifyInterval != 0
```
Comment (ahrtr) [inline tests/linearizability/linearizability_test.go:182]: Timed progress notification is always enabled. Users just can set a different interval. So the name `watchProgressNotifyEnabled` is a little misleading.

**Review by ahrtr — APPROVED** [review]
Overall looks good to me with a minor comment, which can be resolved in a separate PR. 

Merging this PR for now, and I will add a new test case on top of this one.

File: tests/linearizability/linearizability_test.go (hunk 2 above)
Comment (logicalhan) [reply tests/linearizability/linearizability_test.go:182]: It's not though, he's comparing the interval to a zero value, and if it is a zero value, timed progress notifications are effectively disabled.

Comment (ahrtr) [reply tests/linearizability/linearizability_test.go:182]: I was saying etcdserver always enables the timed progress notification. Are you talking about the same thing?

Comment (ahrtr) [reply tests/linearizability/linearizability_test.go:182]: Note: etcdserver always triggers the periodical progress notification for each established watch stream, and it sends progress notification for each watch substream if [ProgressNotify](https://github.com/etcd-io/etcd/blob/39d98522d6b7e3ac4463857d9e7db78eb76131b3/api/etcdserverpb/rpc.pb.go#L1995) is `true`.

Comment (serathius) [reply tests/linearizability/linearizability_test.go:182]: @ahrtr is right, will rephrase it.

**Pull request merged**